### PR TITLE
Update pull request template

### PR DESCRIPTION
**Pull request by patricijabrecko**
## Description
This pull request updates the `.github/pull_request_template.md` file to clarify the pull request process.

## Connected Issues
- https://app.devrev.ai/devrev/works/ISS-204340

## Checklist
- [x] Tests added/updated and ran with `npm run test` OR no tests needed.
- [x] Documentation updated and provided a link to PR / new docs OR `no-docs` written: `no-docs`
- [x] Tested airdrop-template linked to this PR.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,30 +1,23 @@
...
-## Craftsmanship, Integrity, and Devil’s Advocacy
-<!-- Mark the appropriate option with an "x" -->
-- [ ] Testing: Negative test cases: null or default values, crash and fault injection tests
-- [ ] Testing: Boundary conditions: rolling upgrades, denial-of-service, etc.
-- [ ] Testing: Fixing a few existing — flaky or permanently-broken — test cases
-- [ ] Observing: Detailed error codes so machines can understand
-- [ ] Observing: Adding superior metrics for future debugging
-- [ ] Observing: Tracing the hairiest pathways for field serviceability
-- [ ] Training: KnowledgeOps update? So AI always works.
+## Checklist
```
Comment (radovanjorgic) [inline .github/pull_request_template.md:13]: My suggestion is to think of checklist as what MUST be done. In my opinion all checks should be checked, otherwise your PR is not good. I don't say you didn't do it here, but checks should be clear and always checked. So as soon as you read it it is clear what (if anything) is required from you.

For docs case, it should be sth like:
- Provided documentation link or documentation PR, or write no-docs:
For tests maybe: 
- Added tests or no tests needed and verified by running `npm run test`
- Verified backwards compatibility with `npm run test:backwards-compatibility` (once Gasper's PR is merged) 

Not saying you should copy-paste what I suggested but let's try to think what is useful for us together. Also, let's think if there are any other checks needed.

**Review by radovanjorgic — COMMENTED** [review]
Added few comments/suggestions. Also, what I suggest is to use more or less the same PR template structure for all ADaaS repos if possible.

File: .github/pull_request_template.md (hunk 1 above)
Comment (patricijabrecko) [reply .github/pull_request_template.md:12]: I really like this point. I updated the checkboxes to all be written in this way and added another that you should always check your changes with airdrop template.
